src/comm/aca_grpc.cpp

```diff
@@ -0,0 +1,94 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <memory>
+#include <iostream>
+#include <string>
+#include <thread>
+
+#include <grpcpp/server.h>
+#include <grpcpp/server_builder.h>
+#include <grpcpp/server_context.h>
+#include "goalstateprovisioner.grpc.pb.h"
+#include "aca_comm_mgr.h"
+#include "aca_log.h"
+#include "aca_grpc.h"
+
+static char GRPC_SERVER_ADDRESS[] = "0.0.0.0:50001";
+
+using namespace alcor::schema;
+using aca_comm_manager::Aca_Comm_Manager;
+
+Status GoalStateProvisionerImpl::PushNetworkResourceStates(ServerContext *context,
+                                                           const GoalState *goalState,
+                                                           GoalStateOperationReply *goalStateOperationReply)
+{
+  GoalState received_goal_state = *goalState;
+
+  int rc = Aca_Comm_Manager::get_instance().update_goal_state(
+          received_goal_state, *goalStateOperationReply);
+  if (rc == EXIT_SUCCESS) {
+    ACA_LOG_INFO("Control Fast Path synchronized - Successfully updated host with latest goal state %d.\n",
+                 rc);
+  } else if (rc == EINPROGRESS) {
+    ACA_LOG_INFO("Control Fast Path synchronized - Update host with latest goal state returned pending, rc=%d.\n",
+                 rc);
+  } else {
+    ACA_LOG_ERROR("Control Fast Path synchronized - Failed to update host with latest goal state, rc=%d.\n",
+                  rc);
+  }
+
+  return Status::OK;
+}
+
+Status GoalStateProvisionerImpl::PushNetworkResourceStatesStream(
+        ServerContext *context, ServerReaderWriter<GoalStateOperationReply, GoalState> *stream)
+{
+  GoalState goalState;
+  GoalStateOperationReply gsOperationReply;
+  int rc = EXIT_FAILURE;
+
+  while (stream->Read(&goalState)) {
+    rc = Aca_Comm_Manager::get_instance().update_goal_state(goalState, gsOperationReply);
+    if (rc == EXIT_SUCCESS) {
+      ACA_LOG_INFO("Control Fast Path streaming - Successfully updated host with latest goal state %d.\n",
+                   rc);
+    } else if (rc == EINPROGRESS) {
+      ACA_LOG_INFO("Control Fast Path streaming - Update host with latest goal state returned pending, rc=%d.\n",
+                   rc);
+    } else {
+      ACA_LOG_ERROR("Control Fast Path streaming - Failed to update host with latest goal state, rc=%d.\n",
+                    rc);
+    }
+    stream->Write(gsOperationReply);
+    gsOperationReply.Clear();
+  }
+
+  return Status::OK;
+}
+
+void GoalStateProvisionerImpl::RunServer()
+{
+  ServerBuilder builder;
+  builder.AddListeningPort(GRPC_SERVER_ADDRESS, grpc::InsecureServerCredentials());
```

Comment:
Does gRCP server builder support customized heartbeat interval, connection timeout etc.?

Reply:
Looks like many of the connection settings are driven from the client side. Yes, we should look into it when we finalize the communication model from few DPMs (gRPC client) - a lot of ACAs (gRPC server). 

There is client side deadline settings:
https://stackoverflow.com/questions/47646339/grpc-client-side-timeout
https://github.com/grpc/grpc/issues/3954
https://grpc.io/blog/deadlines/

Connection status/keepalive settings:
https://stackoverflow.com/questions/57923846/check-stream-connection-status-grpc
https://stackoverflow.com/questions/57930529/grpc-connection-use-keepalive-or-idletimeout

@haboy52581 please review the above for the connection options for DPM.